### DISCUSSION TITLE
Swap timestamps for SMPTE standard timecode

### DISCUSSION THREAD

**Discussion by peregrineb** [openai/whisper#1214] [Q&A]
I am a documentary filmmaker and I just used Whisper to generate excellent transcripts of my interviews for a small project. The trouble is when Whisper Transcription generates a .srt file, it only gives absolute running-time timecode for each sentence it transcribes. It would be far more useful to me to be able to set a beginning timecode based on the SMPTE timecode associated with the video (or allow Whisper to read the timecode that's available as metadata in the audio file), so that I could jump around in the transcript and pull interview bites I like from the video without having to stop and figure out how far I am from the start point of the transcription.

So as an example, here's a fragment of an interview I transcribed with Whisper, asking for an .srt file:

"1
00:00:00,000 --> 00:00:12,000
[ Inaudible ]
2
00:00:12,000 --> 00:00:13,600
>> Seeing Chris, take one.
3
00:00:13,600 --> 00:00:14,040
Mark.
4
00:00:14,040 --> 00:00:21,120
>> Welcome, welcome, welcome.
5
00:00:21,120 --> 00:00:23,380
You beautiful man.
6
00:00:23,920 --> 00:00:26,180
(laughing)
7
00:00:26,180 --> 00:00:32,500
- Can you please talk to me about your family?"

**What I'd like is a program or script that will take this and turn it into (using a hypothetical start timecode of 15:01:00:00):**

15:01:00 --> 15:01:12
[ Inaudible ]

15:01:12 --> 15:01:13
>> Seeing Chris, take one.

15:01:13 --> 15:01:14
Mark.

15:01:14 --> 15:01:21
>> Welcome, welcome, welcome.

15:01:21 --> 15:01:23
You beautiful man.

15:01:23--> 15:01:26
(laughing)

15:01:26 --> 15:01:32
- Can you please talk to me about your family?

**Could someone easily generate a feature in the end-user program that would allow this?**

**Comment by bbgdzxng1** [comment]
A minor challenge is that a _true_ SMPTE timecode is of the format `HH:MM:SS;ff` or `HH:MM:SS:ff`, and whisper is an audio (rather than video) stream processor, so one would need to pass in an assumed framerate of one of [ 60, 60000/1001, 50, 48, 30, 30000/1001, 25, 24, 24000/1001 ] in order to generate a _true_ SMPTE code and convert `HH:MM:SS.mmm` to `HH:MM:SS;ff` accurately.

There are further challenges of dropframe (DF) vs non dropframe (NDF).

Neither are impossible challenges, but a true implementation needs to consider these two.  True SMPTE timecode would be a prerequisite for SCC and MCC caption support.

**Reply by peregrineb** [reply]
It's true that for 30 fps video the dropframe vs. non-drop can screw things up by several seconds. These days most footage is shot in 24 fps. But for a transcript the last "ff" figure is really irrelevant. I just need to be able to read and seach in the content of an interview, see that I want a sentence (or even a single word) at eg 15:34:18 and go right there in the footage so I can watch/listen and see if it will work.

**Comment by glangford** [comment]
This project has a *fixed-timeshift* utility that may be helpful (I have used *mux* but haven't tested the time shifting). fyi

https://github.com/cdown/srt/blob/develop/srt_tools/README.rst

**Reply by peregrineb** [reply]
That would be very useful as I could probably put in a start SMPTE code by looking at the video or audio footage in the edit program and then apply it.
